app/src/main/java/com/plenart/organizeme/activities/MainActivity.kt

```diff
@@ -3,188 +3,95 @@ package com.plenart.organizeme.activities
 
-class MainActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedListener {
+class MainActivity : AppCompatActivity() {
 
-    private lateinit var mainActivityBinding: ActivityMainBinding;
-    private lateinit var appBarMainBinding: AppBarMainBinding;
-    private lateinit var mainContentBinding: MainContentBinding;
-
-    private lateinit var mUserName: String;
+    private lateinit var binding: ActivityMainBinding
+    private val viewModel: MainActivityViewModel by viewModels()
 
+    private lateinit var toolbar: Toolbar
+    private lateinit var navController: NavController
 
     override fun onCreate(savedInstanceState: Bundle?) {
+        setTheme(R.style.AppTheme)
         super.onCreate(savedInstanceState)
-        mainActivityBinding = ActivityMainBinding.inflate(layoutInflater)
-        setContentView(mainActivityBinding.root)
-
-        setUpActionBar();
-        mainActivityBinding.navView.setNavigationItemSelectedListener(this);
-
-        Firestore().loadUserData(this,true);
 
-        appBarMainBinding.fabCreateBoard.setOnClickListener{
-            val intent = Intent(this,CreateBoardActivity::class.java);
-            intent.putExtra(Constants.NAME, mUserName);
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
 
-            startActivityForResult(intent, CREATE_BOARD_REQUEST_CODE);
-        }
-
-    }
+        toolbar = binding.toolbarMainActivity
+        setSupportActionBar(toolbar)
 
+        setupNavController()
+        setupActionBarWithNavController(navController)
 
-    private fun setUpActionBar(){
-        appBarMainBinding = mainActivityBinding.appBarMainIncluded;
-
-        setSupportActionBar(appBarMainBinding.toolbarMainActivity)
-        appBarMainBinding.toolbarMainActivity.setNavigationIcon(R.drawable.ic_action_navigation_menu);
-
-        appBarMainBinding.toolbarMainActivity.setNavigationOnClickListener {
-            toggleDrawer();
-        }
+        Handler(Looper.getMainLooper()).postDelayed({
+            var currentUserID = Firestore().getCurrentUserID()
+            if (currentUserID.isNotEmpty()) {
+                navController.navigate(R.id.action_splashFragment_to_secNavHostFragment)
+            } else {
+                navController.navigate(R.id.action_splashFragment_to_introFragment)
+            }
+        }, 2500)
 
     }
 
-    private fun toggleDrawer(){
-        if(mainActivityBinding.drawerLayout.isDrawerOpen(GravityCompat.START)){
-            mainActivityBinding.drawerLayout.closeDrawer(GravityCompat.START);
-        }
-        else{
-            mainActivityBinding.drawerLayout.openDrawer(GravityCompat.START);
-        }
-    }
+    private fun setupNavController() {
+        val navHostFragment =
+            supportFragmentManager.findFragmentById(R.id.main_content_navigation_component) as NavHostFragment
+        navController = navHostFragment.navController
```

Comment:
Don't keep reference to nav controller

Reply:
So then, I guess the(more) correct way is to keep a reference to navHostFragment and then when needed access navController as `navHostFragment.navController` ?

Reply:
Well, no. Best way would be to not keep reference at all and use `findNavController` whenever you need access to it. 


But,

as you are here currently using `FragmentContainerView` which actually  complicates `findNavController` a lot, this is ok approach. Only thing I would maybe change is to not keep reference at all and have getter for this.



```
private val mainNavController get() = 
 (supportFragmentManager.findFragmentById(
      R.id.main_content_navigation_component) as?
NavHostFragment)?.findNavController()
```